src/main/java/io/fabric8/maven/docker/config/ConfigHelper.java

```diff
@@ -72,14 +74,35 @@ public static void validateExternalPropertyActivation(MavenProject project, List
             return;
         }
 
-        if(images.size() > 1) {
-            throw new MojoFailureException("Configuration error: Cannot use property "+EXTERNALCONFIG_ACTIVATION_PROPERTY+" on projects with multiple images.");
+        if(images.size() == 1)
+            return;
+
+        // With more than one image, externally activating propertyConfig get's tricky. We can only allow it to affect
+        // one single image. Go through each image and check if they will be controlled by default properties.
+        // If more than one image matches, fail.
+        int imagesWithoutExternalConfig = 0;
+        for (ImageConfiguration image : images) {
+            if(PropertyConfigHandler.canCoexistWithOtherPropertyConfiguredImages(image.getExternalConfig()))
+                continue;
```

Comment:
can we please agree on adding brackets even around single line conditionals ? This is according to the existing style and less error-prone (believe, I hit that when adding a second line).

Also, would prefer to have spaces between `if(` but I'm less dogmatic here ;-)

Reply:
Gah, I thought I configured IntelliJ to do that...